helm/generic-node/templates/generic-node.yaml

```diff
@@ -0,0 +1,301 @@
+        volumeMounts:
+        - mountPath: "/keys/"
+          name: private-keys
+          readOnly: true
+        - mountPath: /wallet-keys
+          name: wallet-keys
+      {{- end }}
+      containers:
+      - name: daemon
+        image: "{{.Values.image.repository }}:{{.Values.image.tag}}"
+        imagePullPolicy: {{.Values.image.pullPolicy}}
+        {{- if .Values.features.customEntrypoint.enabled}}
+        command: [ {{ .Values.features.customEntrypoint.command}} ]
+        {{- end }}
+        volumeMounts:
+        - name: destination-libp2p
+          mountPath: /root/libp2p-keys
+          readOnly: true
+        resources:
+          limits:
+          requests:
+            memory: {{.Values.resources.requests.memory}}
+            cpu: {{.Values.resources.requests.cpu}}
+            rayonThreads: {{.Values.resources.requests.rayonThreads}}
+        args:
+        - "daemon"
+        - "--log-level"
+        - "Debug"
+        - "--log-json"
+        - "--client-port"
+        - "$(DAEMON_CLIENT_PORT)"
+        - "--rest-port"
+        - "$(DAEMON_REST_PORT)"
+        - "--external-port"
+        - "$(DAEMON_EXTERNAL_PORT)"
+        - "--metrics-port"
+        - "$(DAEMON_METRICS_PORT)"
+        - "$(CODA_SNARK_KEY)"
+        - "-snark-worker-fee"
+        - "$(CODA_SNARK_FEE)"
+        - "-work-selection"
+        - "$(WORK_SELECTION)"
+        - "-u"
+        - "/code/watchdog.py"
+        - "/code/make_report.py" 
+        - "--incluster" 
+        - "true"
+        - "--namespace"
+        - ""
+        {{- if .Values.libp2p.staticKeypair.enabled}}
+        - "--discovery-keypair"
+        - "/root/libp2p-keys/key"
+        {{- end }}
+        {{- if .Values.libp2p.seed.enabled}}
+        - "--seed"
+        {{- end }}
+        {{- if .Values.features.runtimeConfig.enabled}}
+        - "--config-file"
+        - "/config/daemon.json"
+        {{- end }}
+        {{- if .Values.features.maxConnections}}
+        - "--max-connections"
+        - {{.Values.features.maxConnections | quote}}
+        {{- end }}
+        {{- range .Values.features.seedPeers}}
+        - "-peer"
+        - {{ . | quote }}
+        {{- end }}
+        {{- if .Values.features.privateKeySecret.enabled}}
+        - "--block-producer-key"
+        - "/root/wallet-keys/key"
+        {{- end }}
+        {{-  if .Values.features.uploadBlocksToGCloud.enabled}}
+        - "-upload-blocks-to-gcloud"
+        - "true"
+        {{- end }}
+        {{-  if .Values.features.persistence.enabled}}
+        - "--config-directory"
+        - "{{.Values.features.persistence.mountPath}}/.mina-config"
+        {{- end }}
+        {{-  if .Values.features.seedPeersLists.enabled}}
+        - "--peer-list-url"
+        - "https://storage.googleapis.com/seed-lists/{{.Values.testnetName}}_seeds.txt"
+        {{- end }}
+        {{-  if .Values.daemon.args}}
+        {{- for args in list}}
```

Comment:
This is not actually proper syntax for a loop but i don't know it off the top of my head